### PR TITLE
Dataloader

### PR DESCRIPTION
**Pull request by remy-byte**
Hello! I need a review on the code I've written.

### CODE REVIEW

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: this won't use the validation dataset at all, according to the documentation at stage == 'fit' you should load both training and validation dataset in order to work properly

Comment (walentines) [reply Data/DataLoader_mask2.py]: then you would have a separate if for stage == 'test' like you already did

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: specifications for every parameter should be added

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: transforms should be added (mean and std should be calculated)

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: you don't have to initialize train_ds, val_ds and test_ds with None here, you could use some dict and do something like stage_ds = dict() and then in setup replace self.train_ds with stage_ds['train']. It's better to use a data structure when you have to store multiple variables.

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: specifications are welcome!

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: I like that you imported the types but didn't use them in methods

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: why double import?

File: Data/test_dataloader.ipynb
Comment (walentines) [inline Data/test_dataloader.ipynb:1]: should be added to .gitignore since we don't want to store it on the repository (it's created automatically by python)

**Review by walentines — CHANGES_REQUESTED** [review]
Please work more on the code since it's not ready yet for the main branch and also create a dir named 'test' where we can write the tests with pytest as we did for DatasetGeneration in order to check if everything is as it should thanks!

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: Args:

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: you don t need to add the type since it s added in init

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: no need of initialization here, just self.stage_ds = {}

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: this looks better now

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: -> TEST_DATALOADERS

File: Data/DataLoader_mask2.py
Comment (walentines) [inline Data/DataLoader_mask2.py]: I would appreciate if this is added before the merge since it's a crucial step in the training (we cannot start the training if we don't have the transforms)

**Review by walentines — COMMENTED** [review]
Changes should still be made to the file but it's way better now! Also I would not approve the merge if the transforms are not properly initialized, we shouldn't neglect that.

**Comment by walentines** [comment]
I would really appreciate if you would take a look at the DatasetGeneration code and try to use the same way of specifications and write tests.

File: Data/test/test_dataloader.py
Comment (walentines) [inline Data/test/test_dataloader.py]: remove this print please!

**Review by walentines — CHANGES_REQUESTED** [review]
Please remove that print so I can approve the merge!

**Review by walentines — APPROVED** [review]
LGTM!

File: Data/test/test_dataloader.py
Comment (remy-byte) [reply Data/test/test_dataloader.py]: okay

**Pull request merged**
